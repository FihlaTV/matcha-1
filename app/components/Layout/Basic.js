import React, { type Node } from 'react';
import { connect } from 'react-redux';
import { StatusBar, View } from 'react-native';
import Header from 'app/components/Header';
import Nav from 'app/components/Nav';
import Toast from 'app/components/Layout/Toast';
import Loader from 'app/components/Layout/Loader';
import { selectToastIsShow, selectToastMessage, selectIsLoading } from 'app/components/Layout/selector';

type Props = {|
  children: Node,
  hasBack?: boolean,
  headerActions?: Array<Node>,
  isLoading: boolean,
  noTabs?: boolean,
  title?: string,
|};

const Layout = (props: Props) => (
  <View style={{ flex: 1 }}>
    <StatusBar />
    <Header hasBack={props.hasBack} title={props.title} actions={props.headerActions} />
    <View style={{ flex: 1 }}>
      {props.children}
    </View>
    { !props.noTabs && <Nav />}
    <Toast
      message={props.toastMessage}
      isShow={props.toastIsShow}
    />
    <Loader isLoading={props.isLoading} />
  </View>
);

Layout.defaultProps = {
  hasBack: false,
  headerActions: [],
  noTabs: false,
  title: undefined,
};

const mapStateToProps = (state) => ({
  isLoading: selectIsLoading(state),
  toastIsShow: selectToastIsShow(state),
  toastMessage: selectToastMessage(state),
});

export default connect(mapStateToProps)(Layout);
