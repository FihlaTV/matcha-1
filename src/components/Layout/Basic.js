import React, { type Node } from 'react';
import { connect } from 'react-redux';
import { StatusBar, View, KeyboardAvoidingView } from 'react-native';
import Header from 'src/components/Header';
import Nav from 'src/components/Nav';
import Toast from 'src/components/Layout/Toast';
import Loader from 'src/components/Layout/Loader';
import { selectToastIsShow, selectToastMessage, selectIsLoading } from 'src/components/Layout/selector';
import { HEADER_HEIGHT } from 'src/constants/design';

type Props = {|
  children: Node,
  hasBack?: boolean,
  headerActions?: Array<Node>,
  isLoading: boolean,
  noTabs?: boolean,
  title?: string,
|};

const Layout = (props: Props) => (
  <View style={{ flex: 1 }}>
    <StatusBar />
    <Header
      hasBack={props.hasBack}
      title={props.title}
      actions={props.headerActions}
    />
    <KeyboardAvoidingView
      style={{
        alignItems: 'center',
        flex: 1,
        justifyContent: 'center',
        marginTop: HEADER_HEIGHT,
      }}
      behavior="padding"
    >
      {props.children}
    </KeyboardAvoidingView>
    { !props.noTabs && <Nav />}
    <Toast
      message={props.toastMessage}
      isShow={props.toastIsShow}
    />
    <Loader isLoading={props.isLoading} />
  </View>
);

Layout.defaultProps = {
  hasBack: false,
  headerActions: [],
  noTabs: false,
  title: undefined,
};

const mapStateToProps = (state) => ({
  isLoading: selectIsLoading(state),
  toastIsShow: selectToastIsShow(state),
  toastMessage: selectToastMessage(state),
});

export default connect(mapStateToProps)(Layout);
