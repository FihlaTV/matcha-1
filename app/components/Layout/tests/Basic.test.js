import React from 'react';
import Basic from 'app/components/Layout/Basic';
import BasicWeb from 'app/components/Layout/Basic.web';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

describe('Layout', () => {
  describe('render test', () => {
    it('should render', () => {
      const wrapper = shallow(<Basic />);
      expect(toJson(wrapper)).toMatchSnapshot();
      const wrapperWeb = shallow(<BasicWeb />);
      expect(toJson(wrapperWeb)).toMatchSnapshot();
    });
  });
});
