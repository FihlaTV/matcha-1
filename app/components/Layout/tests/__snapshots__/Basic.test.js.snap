// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout render test should render 1`] = `
<Layout
  hasBack={false}
  headerActions={Array []}
  noTabs={false}
  store={Object {}}
/>
`;

exports[`Layout render test should render 2`] = `
<Layout
  noTabs={false}
  store={Object {}}
  style={Object {}}
/>
`;
